src/source_sampling.cpp

```diff
@@ -179,6 +188,17 @@ void pyne::Sampler::setup() {
   }
   else throw std::invalid_argument("Mesh file must contain only tets or hexes.");
 
+  // Assign MeshMode: VOXEL, SUBVOXEL, TET
+  if (ve_type == moab::MBHEX){
+     if (mode < 3){
```

Comment:
Make `mode` a `MeshMode` enumerator value and you won't have to this extra check.